### PR TITLE
Engage pages tags filter

### PR DESCRIPTION
**Pull request by carkod**
## Done

- Tags filter for engage pages
- Show tags on each card

## QA

- Check https://ubuntu-com-10192.demos.haus/engage
- Use the tags filter

## Issue / Card

Fixes https://github.com/canonical-web-and-design/web-squad/issues/4325
Fixes https://github.com/canonical-web-and-design/web-squad/issues/3874

### CODE REVIEW

**Comment by pmahnke** [comment]
@carkod and @davegoddard42 I think it is pretty easy to select a state where there are no engage pages, what about a reset?

image

**Comment by davegoddard42** [comment]
@pmahnke @carkod Reset sounds good. It also needs a 'empty state' for when there's no content. I can draw that this afternoon for you, @carkod 

Depending on how much time/effort we want to spend on this, we could also revise the use of drop downs, and bring it in line with the 'Search and Filter' pattern and Pagination pattern we have for items like these. Probably beyond the scope of this ticket.

**Comment by davegoddard42** [comment]
@pmahnke @carkod 

I'm looking at this further, and I've realised that drop downs will only allow you to select one parameter per category. Is that intentional, or a limitation of the pattern chosen? The current site has only a language dropdown. That sort of makes sense, I might only want to see documents in one language. But I don't know if using drop downs for the other items is right?

**Comment by pmahnke** [comment]
I _think_ it would be ok for now.  There isn't that much content.

Perhaps in the future we look at a richer solution, like you mentioned?

**Comment by davegoddard42** [comment]
@carkod I've done a design with the filters added. I also mocked up what it could look like using our S&F patterns, but as per Peter's comments above, we won't do that now.

https://miro.com/app/board/o9J_l2GS0J8=/

**Comment by carkod** [comment]
> @carkod I've done a design with the filters added. I also mocked up what it could look like using our S&F patterns, but as per Peter's comments above, we won't do that now.
> 
> https://miro.com/app/board/o9J_l2GS0J8=/

Thanks. Done.

**Comment by davegoddard42** [comment]
LGTM @carkod

File: templates/engage/index.html [hunk 1]
```diff
@@ -19,56 +19,101 @@
...
+          <option value="all">All tags</option>
+          {% for tag in tags %}
+            <option value="{{ tag }}">{{ tag }}</option>
+          {% endfor %}
+        </select>
+      </div>
+    </div>
+    {% endif %}
+    <div class="col-3" style="display: flex; align-items: flex-end; justify-content: flex-end;">
+      <button id="clear-filters" class="u-no-margin--bottom" onclick="clearFilters()" type="reset" disabled>Clear filters</button>
```
Comment (anthonydillon) [inline templates/engage/index.html:70]: Could you add a `<br class="u-hide u-show--small" />` above this button to add some space on small screens:
ubuntu-com-10192 demos haus_engage_language=en resource=whitepaper tag=maas

File: templates/engage/index.html [hunk 2]
```diff
@@ -19,56 +19,101 @@
...
+        type=item.type,
+        preview=preview,
+        active=item.active,
+        tags=item.tags
+      %}
       {% include "engage/_article-card.html" %}
       {% endwith %}
     {% endfor %}
-
+    <p></p>
```
Comment (anthonydillon) [inline templates/engage/index.html:92]: Empty p?

File: templates/engage/index.html [hunk 3]
```diff
@@ -19,56 +19,101 @@
...
   </div>
-
+  {% else %}
+  <div class="u-fixed-width">
+    <h3>No results - why not try widening your search?</h3>
```
Comment (anthonydillon) [inline templates/engage/index.html:96]:
```suggestion
    <h3>No results &mdash; try widening your search?</h3>
```

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few small requests in the code

File: templates/engage/index.html [hunk 4]
```diff
@@ -19,56 +19,101 @@
...
+          <option value="all">All tags</option>
+          {% for tag in tags %}
+            <option value="{{ tag }}">{{ tag }}</option>
+          {% endfor %}
+        </select>
+      </div>
+    </div>
+    {% endif %}
+    <div class="col-3" style="display: flex; align-items: flex-end; justify-content: flex-end;">
+      <br class="u-hide u-show--small" />
```
Comment (anthonydillon) [inline templates/engage/index.html:70]:
```suggestion
      <div class="u-hide u-show--small" />
          <br/><br/>
      </div>
```

**Review by anthonydillon — APPROVED** [review]
LGTM :+1: with a small suggestion

File: templates/engage/index.html (hunk 4 above)
Comment (carkod) [reply templates/engage/index.html:70]: I suppose you mean only 1 `<br />`? What's the purpose of using 2?

Comment (carkod) [reply templates/engage/index.html]: Done, it's aligning now

**Pull request merged**